automations/python/workflows/get_folder_differences.py

```diff
@@ -0,0 +1,143 @@
+"""
+Determine which files have changed in the documentation preview between
+the preview branch and the main branch.
+
+This uses the `diff` CLI utility to determine the list of files that have
+changed. It then converts the paths to URLs that can be used to view the
+files on the docs preview site. Finally, it outputs the preview site docs
+generated comment along with a list of new and changed files.
+"""
+import os
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+
+from shared.actions import write_to_github_output
+
+
+ORIGINAL_FOLDER = Path("/tmp/gh-pages")
+OUTPUT_FOLDER = Path("/tmp/gh-pages-for-diff/")
+BASE_URL = "https://docs.openverse.org"
+PR_NUMBER = os.environ.get("PR_NUMBER")
+EXCLUSIONS = {
+    "*.js",
+    "*.json",
+    "*.html",
+    ".*",
+    "CNAME",
+    "*.pickle",
+    "*.doctree",
+    "*.inv",
```

Comment:
```suggestion
    "*.inv",
    "*.txt",
    "*.map",
```

These two also need to be excluded based on the comment to this PR. I'm not so sure about the `txt`, though, would there ever be `txt` files added that we would like to review?

Reply:
Would it be easier to only include relevant files, instead of playing whack-a-mole with file types?

https://stackoverflow.com/a/10132592

Reply:
@sarayourfriend I've tried a few different attempts based off this approach ([example](https://github.com/WordPress/openverse/actions/runs/5906329361/job/16022325010?pr=2647)) but for some reason `diff` is _still_ comparing files it should be excluding 😕 it's not ideal, but in the interest of not spending hours on this I'll add the additional exclusions and keep the current approach.

Reply:
I've also added some handling in 8e5ff36e19fec512ace39beb90ad4b4737cf9763 to ensure we only display valid HTML pages in the diff.